starknet_py/utils/crypto/facade.py

```diff
@@ -1,19 +1,84 @@
+import functools
 import os
-from typing import Optional
+from typing import Optional, Sequence, Callable
 
 from crypto_cpp_py.cpp_bindings import (
     cpp_hash,
     get_cpp_lib_file,
     ECSignature,
 )
-from starkware.cairo.common.hash_state import compute_hash_on_elements
 from starkware.cairo.lang.vm.crypto import pedersen_hash as default_hash
 from starkware.crypto.signature.signature import sign
 
+from starknet_py.common import int_from_bytes
 from starknet_py.net.client_models import Call
 
-# PREFIX_TRANSACTION = encoded 'StarkNet Transaction'
-PREFIX_TRANSACTION = 476441609247967894954472788179128007176248455022
+# From cairo-lang
+CONTRACT_ADDRESS_PREFIX = int_from_bytes(b"STARKNET_CONTRACT_ADDRESS")
+L2_ADDRESS_UPPER_BOUND = 2**251 - 256
```

Comment:
I've previously put constants like these in the `constants.py` file.
I know these probably won't be used anywhere else.
But after all the file `constants.py` is supposed to group and hold constants, so idk.